packages/amplify-category-auth/API.md

```diff
@@ -86,6 +86,11 @@ export function externalAuthEnable(context: any, externalCategory: any, resource
 // @public (undocumented)
 export const generateAuthStackTemplate: (context: $TSContext, resourceName: string) => Promise<Template>;
 
+// Warning: (ae-forgotten-export) The symbol "AuthTriggerCfnTypes" needs to be exported by the entry point index.d.ts
+//
+// @public (undocumented)
+export const getAuthTriggerStackCfnParameters: (context: $TSContext, authResourceName: string) => Promise<AuthTriggerCfnTypes>;
+
```

Comment:
does this have to be public ?

if yes, then can this become more generalized concept? 
From the usage I can imagine that we might want to have abstraction which works for evert category in case we want to use it in nested stack.
For example
`getStackCfnParameters(context, resourceName): Promise<something_more_generic>` - this option fixes naming making it a possible pattern.
`appendCfnStackParameters(stack, context, resourceName): Promise<void>` - this would mean that how category appends paramteres becomes internal details (aka visitor pattern).

Reply:
The logic is specific to auth triggers , thats why I added it in the stack. 

Above pattern is useful if we are refactoring this logic to get all CFN parameters from each categories and in that flow we can call ```getAuthTriggerStackCfnParameters ```  to get parameters for trigger stack , inside ```appendCfnStackParameters ()```.

I have to refractor ```cognito``` stack logic as well since both ```cognito trigger stack``` and ```cognito stack```  both comes auth category.

We can work on above when we are refactoring push-resources logic.